src/library-authoring/components/ComponentDeleter.tsx

```diff
@@ -62,40 +51,40 @@ const ComponentDeleter = ({ usageKey, ...props }: Props) => {
         onClick: restoreComponent,
       },
     );
-    props.cancelDelete();
+    close();
     // Close the sidebar if it's still open showing the deleted component:
     if (usageKey === sidebarComponentUsageKey) {
       closeLibrarySidebar();
     }
   }, [usageKey, sidebarComponentUsageKey, closeLibrarySidebar]);
 
-  const { hits } = useContentFromSearchIndex([usageKey]);
+  const { hits, isLoading } = useContentFromSearchIndex([usageKey]);
   const componentHit = (hits as ContentHit[])?.[0];
 
-  if (!props.isConfirmingDelete) {
+  // istanbul ignore if: loading state
+  if (isLoading || isLoadingLinks) {
```

Comment:
Previously, the data was being filled as it arrived. Only rendering after we get all the data prevents some wiggle in the dialog.